### PR TITLE
Feature/update animal table

### PR DESCRIPTION
**Pull request by M-Anwar-Hussaini**
# New Commands in the Milestone:

## Add a new column by the name of `species` to the animal table.
image

## Insert some new data to the table:
image
image

## Create Several transactions:
image
image
image

## Fetch data from database using aggregation functions:
image
image
image
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -21,3 +21,71 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+-- ---------------- --
+-- SECOND MILESTONE --
+-- ---------------- --
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (rloterh) [inline queries.sql:35]: - [ ] As a good practice, let's ensure to verify that changes are verified before and after rollback.

File: queries.sql [hunk 2]
```diff
@@ -21,3 +21,71 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+-- Verify that changes were made.
+-- Commit the transaction.
+-- Verify that changes persist after commit.
+
+BEGIN;
+UPDATE animals SET species = 'digimon ' WHERE NAME LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+SELECT * FROM animals ORDER BY id;
```
Comment (rloterh) [inline queries.sql]: - [ ] Similarly, let's verify that changes are verified before and after commit.

File: queries.sql [hunk 3]
```diff
@@ -21,3 +21,71 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+
+
+-- Delete all records in the animals table, then roll back the transaction.
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (rloterh) [inline queries.sql]: - [ ] Again, as a good practice, let's verify that changes are verified before and after rollback. Our query should look like:

```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

File: queries.sql [hunk 4]
```diff
@@ -21,3 +21,71 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
...
+WHERE escape_attempts > 0 GROUP BY neutered ORDER BY escape_count DESC;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MAX(weight_kg) AS max_weight, MIN(weight_kg) AS min_weight
+FROM animals GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2010?
+SELECT species, AVG(escape_attempts) AS average_escape_attempts FROM animals
+WHERE date_of_birth BETWEEN '1990-1-1' AND '2010-1-1' GROUP BY species;
+
```
Comment (rloterh) [inline queries.sql]: - [ ] Good job, let's however note that the scenario that we are looking for over here is **_What is the average number of escape attempts per animal type of those born between 1990 and 2000?_** and the query should look like:

```sql
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31';
```

Let's endeavor to improve upon the same.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Anwar, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 __🏃 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍  
- Good job so far implementing update queries!
- ReadMe looks professional!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (M-Anwar-Hussaini) [reply queries.sql]: Hi @rloterh, Thanks a lot for providing your feedback.

The code you have suggested doesn't work and gives the error:
image

I guess the second and third lines should be swapped.

**Review by KDlamini — APPROVED** [review]
### Hi @M-Anwar-Hussaini 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for querying and updating animals table for the Vet clinic database. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  Professional pull request
✔️  Correct use of Gitflow 
✔️  All required SQL commands are in their respective files

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**